### ISSUE TITLE
[Bug]:  "FusionAuth login page" don't display result.errors.generalErrors returned by the "Login Validation Lambda" when using an identity provider to log in

### ISSUE THREAD

**Issue by konvergence** [FusionAuth/fusionauth-issues#2903]
When you have a login validation lambda and use an identity provider to log in, you don't see the correct error message. Instead you see 

```
A validation error occurred during the login attempt. An event log was created for the administrator to review. 
```

Replication steps:

* set up a login lambda:
```
// Validate the login request by optionally setting errors in the result.
function validate(result, user, registration, context) {
    if (registration.applicationId === "3769def5-70fd-4984-9695-231f9c05277d") {      
        
            result.errors.generalErrors = [{
              code: "[LoginRestricted]",
              message: "login restricted"
            }];
    }   
}
```

This only prevents users registered to the application with an id of 3769... from logging in.
* Create an application with the id `3769def5-70fd-4984-9695-231f9c05277d`. It can have an authorized redirect URL of https://fusionauth.io. That is, you don't need to actually do the token exchange to see this issue.
* Register a user for this application in the admin UI.
* Create a new theme. 
* Update the messages of this theme to add a message for `[LoginRestricted]`:
```
[LoginRestricted]=you shall not pass
...
```
* Assign the login lambda and the theme to this tenant.
* Create an OIDC provider. I used sandbox.fusionauth.io, federating from a local instance.
  * On the sandbox side
    * create an application. 
    * enable the authorization code grant
    * set the redirect URL to `http://localhost:9011/oauth2/callback`
    * record the client id and client secret
  * On the localhost side
    * create an OIDC identity provider
    * set the client id and client secret to the values recorded in previous step
    * set 'discover endpoints' to true
    * set the issuer to `https://sandbox.fusionauth.io`
    * set scopes to `profile email openid`
    * enable it for the application starting with 3769...
* Open an incognito window.
* Log in with the registered user using username and password
* You'll see the message `you shall not pass`
* Log in with the OIDC Identity provider
* You'll see the validation error message:
```
A validation error occurred during the login attempt. An event log was created for the administrator to review. 
```

Expected:

You'd see the custom error message no matter how you log in.

----

Original issue report.
### What happened?

Hi,

In the doc https://fusionauth.io/docs/extend/code/lambdas/login-validation
 we can define a localized code and a message into result.errors.generalErrors, like this
```
function validate(result, user, registration, context) {
  if (user.data.accountStatus === 'pastDue') {
    result.errors.generalErrors = [{
      code: "[LoginRestricted]",
      message: "Account is past due, please contact accounts payable."
    }];
  }
}
```

When a associate the lamdba into the tenant security
I got well reject, 
<img width="431" alt="2024-10-18_14-30" src="https://github.com/user-attachments/assets/79891260-e65f-4d4b-bfef-dce70ae4c99a">

but I get this message :
 `A validation error occurred during the login attempt. An event log was created for the administrator to review.`

<img width="196" alt="2024-10-18_14-28_1" src="https://github.com/user-attachments/assets/567078d1-8c6c-4234-b2e5-dc7826469338">

And I can not find any message with this text into the messages of the default "FusionAuth" Theme 75a068fd-e94b-451a-9aeb-3ddb9a3b5987

### Version

1.53.2

### Affects Versions

1.53.2

**Comment by mooreds** [comment]
@konvergence Hmmm. 

Can you please provide step by step replication? It's hard for me to understand what is going on based on this bug report.

1. I set up a login lambda with this code: ...
2. I set up a custom theme
3. I updated the message file for the english and french message bundles to include this message: ...
4. I associated the login lambda with the tenant
5. I opened up an incognito browser and then logged in.
6. I saw ... on the event log
7. I saw ... on the screen as a user

That level of detail will help us troubleshoot.

Thanks!

**Comment by konvergence** [comment]
Hi  @mooreds 
please find the details steps

1. I set up a login lambda with this code: ...
```javascript
// Validate the login request by optionally setting errors in the result.
function validate(result, user, registration, context) {
    if (registration.applicationId === "3769def5-70fd-4984-9695-231f9c05277d") {      
        if ( user.data.roles &&  user.data.roles.indexOf("R-XGD1-KCSR_Acces_CSR_Insight") < 0 ) {
            result.errors.generalErrors = [{
              code: "[LoginRestricted]",
              message: "login restricted"
            }];
        }
    }   
}
```
2. I set up a custom theme
3. I updated the message file for the english and french message bundles to include this message: ...
4. <img width="516" alt="2024-10-22_08-29" src="https://github.com/user-attachments/assets/a7aa4a85-41b8-4213-90cd-fd94955a1521">

5. I associated the login lambda and theme with the tenant
<img width="457" alt="2024-10-22_08-32" src="https://github.com/user-attachments/assets/ba08beba-a9bf-443b-807a-572282e0a67c">
<img width="467" alt="2024-10-22_08-34" src="https://github.com/user-attachments/assets/0293718c-a2b7-44c4-8b8f-a6945f18de16">
6. I opened up an incognito browser and then logged in.
<img width="322" alt="2024-10-22_08-37" src="https://github.com/user-attachments/assets/28f64ae1-021a-49a4-be8b-ddaab7b9c001">

7. I saw ... on the event log
<img width="451" alt="2024-10-22_08-40" src="https://github.com/user-attachments/assets/6915ce8b-40b3-4b95-8c0a-c3a04c429c01">

details of Error
<img width="297" alt="2024-10-22_08-42" src="https://github.com/user-attachments/assets/7576b5c6-e158-4ac0-a95a-1b7265ed8b3d">


8. I saw ... on the screen as a user 
<img width="322" alt="2024-10-22_08-37" src="https://github.com/user-attachments/assets/28f64ae1-021a-49a4-be8b-ddaab7b9c001">

**Comment by mooreds** [comment]
Thank you. I was able to replicate the error. I saw the functionality work fine if you log in with a password, but if you try to use an OIDC provider, you see the exception.

**Comment by mooreds** [comment]
@konvergence I determined that this bug appears if you only have one language enabled as well, so updated the description with simpler replication steps.

**Comment by mooreds** [comment]
Thanks!

**Comment by robotdan** [comment]
This one is a little tricky. But I'm sure we can solve it. 

The reason for this behavior is that in general, we expect the login API to behave in a specific way. When it doesn't - it is a failure to connect to the IdP, or something like that. 

In those cases, this is not something the end user needs to know about, it is sort of a runtime error that a developer needs to solve. 

So - for these reasons, we treat this as unexpected and tell the user "Sorry, something broke", and log it for the admin to review since we don't expect the user can do anything about it.

With the addition of this lambda, this contract is now more complicated in that we do want to show the end user the validation error that was returned by the API by way of the lambda function. So this was a miss when we shipped this new feature.

We'll just have to figure out how to tell the difference between these two states and correctly return the errors returned by the lambda function instead of returning the generic message you are seeing: 

```
[IdentityProviderValidationException]=A validation error occurred during the login attempt. An event log was created for the administrator to review.
```

**Comment by konvergence** [comment]
Ok
So if i define the message on a theme, this can be a workarround for the ui.

But i think there an issue with oidc provider versus samlv2 provider.

With both providers, on the ui,  i well see the message
[IdentityProviderValidationException]=A validation error occurred during the login attempt. An event log was created for the administrator to review.

With an SAMLv2 provider, i can retry the login, like in step 8.

But with an oidc provider, i can not retry, because there an oauth2 return error.
(Need add a screenshot)

**Comment by robotdan** [comment]
Thanks @konvergence 

We should have a patch out today or tomorrow. 

I tested the workflow with the fix which allows us to show the correct message provided by the lambda during an OIDC or other IdP based login. I correctly see the error and can retry by just hitting the login button again. 

I tested with an OpenID Connect IdP.

**Comment by robotdan** [comment]
@konvergence Fix is in 1.53.3 - let us know if this works for you! 
- https://fusionauth.io/docs/release-notes/
